webapp/channels/src/packages/mattermost-redux/src/selectors/entities/users.ts

```diff
@@ -860,3 +861,38 @@ export const getLastActiveTimestampUnits: (state: GlobalState, userId: string) =
         return timestampUnits;
     },
 );
+
+export const getRemoteClusters = (state: GlobalState) => state.entities.remoteClusters || {};
+
+export const userCanSeeOtherUser: (state: GlobalState, userId: string) => boolean = createSelector(
+    'userCanSeeOtherUser',
+    (state: GlobalState, userId: string) => userId,
+    (state: GlobalState, userId: string) => getUser(state, userId),
+    (state: GlobalState) => state.entities.general.config,
+    getRemoteClusters,
+    (userId, user, config, remoteClusters) => {
+        if (!user) {
+            return false;
+        }
+
+        // Feature flag check
+        const featureFlags = config as any;
+        if (!featureFlags.FeatureFlags || featureFlags.FeatureFlags.EnableSharedChannelsDMs !== 'true') {
+            return true; // Always allow if feature is disabled
+        }
+
+        // Local users (without remote_id) can always be messaged
+        if (!user.remote_id) {
+            return true;
+        }
+
+        // Check if the user's remote cluster is directly connected
+        const remoteCluster = remoteClusters[user.remote_id];
+        if (!remoteCluster) {
+            return false;
+        }
+
+        // A remote is directly connected if it's both confirmed and connected
+        return isConfirmed(remoteCluster) && isConnected(remoteCluster);
```

Comment:
Same question as in the backend file. In this case, the fact that the remote is disconnected shouldn't be an issue to start a DM, right?